staging/src/k8s.io/client-go/rest/client_test.go

```diff
@@ -93,7 +93,7 @@ func TestDoRequestFailed(t *testing.T) {
 		Message: " \"\" not found",
 		Details: &metav1.StatusDetails{},
 	}
-	expectedBody, _ := runtime.Encode(api.Codecs.LegacyCodec(v1.SchemeGroupVersion), status)
+	expectedBody, _ := runtime.Encode(scheme.Codecs.LegacyCodec(v1.SchemeGroupVersion), status)
```

Comment:
The codec we get from `codec := c.LegacyCodec(version)` is going to encode to `version` and decode to `internalVersion`. So using it to encode is fine, but to decode, we need to extract the serializer part or wrap the scheme.Codec into a DirectCodec, which is used by the clientset.